js/viz.js

```diff
@@ -114,5 +132,48 @@ const viz = {
     this.context.closePath();
     this.context.strokeStyle = '#ccc';
     this.context.stroke();
+  },
+
+  isPointInsideCircle(x, y, cx, cy) {
+    const dx = Math.abs(x - cx);
+    const dy = Math.abs(y - cy);
+    const d = dx*dx + dy*dy;
+    const r = this.radius;
+
+    return d <= r*r;
+  },
+
+  getNodeAtCoordinates(x, y) {
+    for (const node of this.nodes) {
+      if (this.isPointInsideCircle(x, y, node.x, node.y)) {
+        return node;
+      }
+    }
+    return null;
+  },
+
+  addListeners() {
+    this.addMouseMove();
+    // this.addMouseLeave();
+  },
+
+  addMouseMove() {
+    this.canvas.on('mousemove', () => {
+      const points = d3.mouse(this.canvas.node());
+      // const X = d3.event.layerX || d3.event.offsetX;
+      const node = this.getNodeAtCoordinates(points[0], points[1]);
+      // const node = this.simulation.find(points[0], points[1], this.radius);
```

Comment:
My hunch is that this would be right if another tick happens. We perhaps need to store currentX and Y of nodes until tick happens. Try calling tick first before calculating and see if it works. I have a feeling we have conflated the meaning of x and y here.